src/functions/projectConfig.ts

```diff
@@ -0,0 +1,50 @@
+import { FunctionsConfig, FunctionsSingle } from "../firebaseConfig";
+import { FirebaseError } from "../error";
+
+export type NormalizedConfig = [FunctionsSingle, ...FunctionsSingle[]];
+export type ValidatedSingle = FunctionsSingle & { source: string };
+export type ValidatedConfig = [ValidatedSingle];
+
+/**
+ * Normalize functions config to return functions config in an array form.
+ */
+export function normalize(config: FunctionsConfig | undefined): NormalizedConfig {
+  if (config === undefined) {
```

Comment:
Does it need to specifically equal undefined?

```suggestion
  if (!config) {
```

Reply:
This is probably fine!